integrations/azure_ai_search/tests/test_document_store.py

```diff
@@ -0,0 +1,365 @@
+    @pytest.fixture
+    def filterable_docs(self) -> List[Document]:
+        """Fixture that returns a list of Documents that can be used to test filtering."""
+        documents = []
+        for i in range(3):
+            documents.append(
+                Document(
+                    content=f"A Foo Document {i}",
+                    meta={
+                        "name": f"name_{i}",
+                        "page": "100",
+                        "chapter": "intro",
+                        "number": 2,
+                        "date": "1969-07-21T20:17:40Z",
+                    },
+                    embedding=_random_embeddings(768),
+                )
+            )
+            documents.append(
+                Document(
+                    content=f"A Bar Document {i}",
+                    meta={
+                        "name": f"name_{i}",
+                        "page": "123",
+                        "chapter": "abstract",
+                        "number": -2,
+                        "date": "1972-12-11T19:54:58Z",
+                    },
+                    embedding=_random_embeddings(768),
+                )
+            )
+            documents.append(
+                Document(
+                    content=f"A Foobar Document {i}",
+                    meta={
+                        "name": f"name_{i}",
+                        "page": "90",
+                        "chapter": "conclusion",
+                        "number": -10,
+                        "date": "1989-11-09T17:53:00Z",
+                    },
+                    embedding=_random_embeddings(768),
+                )
+            )
+
+            documents.append(
+                Document(content=f"Doc {i} with zeros emb", meta={"name": "zeros_doc"}, embedding=TEST_EMBEDDING_1)
+            )
+            documents.append(
+                Document(content=f"Doc {i} with ones emb", meta={"name": "ones_doc"}, embedding=TEST_EMBEDDING_2)
+            )
+        return documents
+
+    # Overriding to compare the documents with the same order
+    def assert_documents_are_equal(self, received: List[Document], expected: List[Document]):
+        """
+        Assert that two lists of Documents are equal.
+
+        This is used in every test, if a Document Store implementation has a different behaviour
+        it should override this method. This can happen for example when the Document Store sets
+        a score to returned Documents. Since we can't know what the score will be, we can't compare
+        the Documents reliably.
+        """
+        sorted_recieved = sorted(received, key=lambda doc: doc.id)
+        sorted_expected = sorted(expected, key=lambda doc: doc.id)
+        assert sorted_recieved == sorted_expected
+
+    # Dataframes are not supported in serach index
+    def test_comparison_equal_with_dataframe(self, document_store, filterable_docs):
```

Comment:
Let's use an explicit skip mark like this
https://github.com/deepset-ai/haystack-core-integrations/blob/e97cfee4c2b0868d96e4a4a335ec5cd7ee0f6cf1/integrations/pinecone/tests/test_filters.py#L41